File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/TaskLinks/predecessorandsuccessortask.rb
module Asposetasksjava
  module PredecessorAndSuccessorTask
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        tsk = Rjb::import('com.aspose.tasks.Tsk')

    	  allinks = project.getTaskLinks().toList()

        i = 0
        while i < allinks.size()
          tsklnk = allinks.get(i)
          puts "Predecessor " + tsklnk.getPredTask().get(tsk.NAME).to_string
          puts "Successor " + tsklnk.getSuccTask().get(tsk.NAME).to_string
          puts "------------------------------------------------------"
          i +=1
        end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTaskLinks/PredecessorAndSuccessorTask.php
<?php
namespace Aspose\Tasks\WorkingWithTaskLinks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Tsk as Tsk;
class PredecessorAndSuccessorTask{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $tsk = new Tsk();

        $allinks = $project->getTaskLinks()->toList();
        $i = 0;
        while ($i < sizeof($allinks)) {

          $tsklnk = $allinks -> get($i);
          print "Predecessor " . (string)$tsklnk -> getPredTask() -> get($tsk-> NAME).PHP_EOL;// . to_string
          print "Successor " . (string)$tsklnk -> getSuccTask() -> get($tsk -> NAME).PHP_EOL;// . to_string
          print "------------------------------------------------------".PHP_EOL;
          $i += 1;
        }
    }

}
?>